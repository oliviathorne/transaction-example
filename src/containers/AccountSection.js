// @flow
import React from 'react';
import styled from 'styled-components';
import TransactionWrapper from './TransactionWrapper';
import AccountSummary from './../components/AccountSummary';

const AccountWrapper = styled.div`
  width: 50%;
  padding: 2rem 0 4rem 10%;
  box-sizing: border-box;

  @media (max-width: 768px) {
    width: 100%;
    padding: 2rem 0 4rem;
  }
`;

const H1 = styled.h1`
  font-family: 'AlverataBlack';
  letter-spacing: -0.6px;
`;

const H3 = styled.h3`
  font-family: 'AlverataBlack';
  letter-spacing: -0.6px;
  margin: 5rem 0 0.5rem 0;
`;

const AccountSection = () => (
  <AccountWrapper>
    <H1>My account</H1>
    <AccountSummary />

    <H3>Transactions</H3>
    <TransactionWrapper />
  </AccountWrapper>
);

export default AccountSection;
