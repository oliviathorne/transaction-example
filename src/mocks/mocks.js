export const mockState = {
  form: {
    fields: {},
    savedTransactions: [
      {
        name: {
          value: 'Test name'
        },
        email: {
          value: 'email@test.com'
        },
        amount: {
          value: '3000'
        }
      }
    ],
    startingBalance: 18000,
    totalAvailable: 18000,
    totalSent: 0
  }
};
