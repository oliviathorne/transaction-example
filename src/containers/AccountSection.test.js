import { shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import AccountSection from './AccountSection';
import { mockState } from './../mocks/mocks';

const mockStore = configureStore();
let store;
let wrapper;

beforeEach(() => {
  store = mockStore(mockState);
  wrapper = shallow(
    <Provider store={store}>
      <AccountSection />
    </Provider>
  );
});

describe('AccountSection component', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
