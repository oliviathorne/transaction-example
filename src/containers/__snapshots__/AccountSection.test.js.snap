// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSection component renders correctly 1`] = `<AccountSection />`;
